modules/platforms/python/cpp_module/utils.cpp

```diff
@@ -0,0 +1,395 @@
+
+    PyObject* u_second = PyLong_FromLong(value.get_nano() / 1000);
+    if (!u_second)
+        return nullptr;
+    auto u_second_guard = ignite::detail::defer([&]{ Py_DECREF(u_second); });
+
+    auto args = PyTuple_Pack(4, hour, minute, second, u_second);
+    if (!args)
+        return nullptr;
+    auto args_guard = ignite::detail::defer([&]{ Py_DECREF(args); });
+
+    hour_guard.release();
+    minute_guard.release();
+    second_guard.release();
+    u_second_guard.release();
+
+    auto kwargs = PyDict_New();
+    if (!kwargs)
+        return nullptr;
+    auto kwargs_guard = ignite::detail::defer([&]{ Py_DECREF(kwargs); });
+
+    return PyObject_Call(time_class, args, kwargs);
+}
+
+PyObject* py_create_datetime(const ignite::ignite_date_time &value) {
+    auto datetime_class = py_get_module_datetime_class();
+    if (!datetime_class)
+        return nullptr;
+    auto class_guard = ignite::detail::defer([&]{ Py_DECREF(datetime_class); });
+
+    PyObject* year = PyLong_FromLong(value.get_year());
+    if (!year)
+        return nullptr;
+    auto year_guard = ignite::detail::defer([&]{ Py_DECREF(year); });
+
+    PyObject* month = PyLong_FromLong(value.get_month());
+    if (!month)
+        return nullptr;
+    auto month_guard = ignite::detail::defer([&]{ Py_DECREF(month); });
+
+    PyObject* day = PyLong_FromLong(value.get_day_of_month());
+    if (!day)
+        return nullptr;
+    auto day_guard = ignite::detail::defer([&]{ Py_DECREF(day); });
+
+    PyObject* hour = PyLong_FromLong(value.get_hour());
+    if (!hour)
+        return nullptr;
+    auto hour_guard = ignite::detail::defer([&]{ Py_DECREF(hour); });
+
+    PyObject* minute = PyLong_FromLong(value.get_minute());
+    if (!minute)
+        return nullptr;
+    auto minute_guard = ignite::detail::defer([&]{ Py_DECREF(minute); });
+
+    PyObject* second = PyLong_FromLong(value.get_second());
+    if (!second)
+        return nullptr;
+    auto second_guard = ignite::detail::defer([&]{ Py_DECREF(second); });
+
+    PyObject* u_second = PyLong_FromLong(value.get_nano() / 1000);
+    if (!u_second)
+        return nullptr;
+    auto u_second_guard = ignite::detail::defer([&]{ Py_DECREF(u_second); });
+
+    auto args = PyTuple_Pack(7, year, month, day, hour, minute, second, u_second);
+    if (!args)
+        return nullptr;
+    auto args_guard = ignite::detail::defer([&]{ Py_DECREF(args); });
+
+    year_guard.release();
+    month_guard.release();
+    day_guard.release();
+    hour_guard.release();
+    minute_guard.release();
+    second_guard.release();
+    u_second_guard.release();
+
+    auto kwargs = PyDict_New();
+    if (!kwargs)
+        return nullptr;
+    auto kwargs_guard = ignite::detail::defer([&]{ Py_DECREF(kwargs); });
+
+    return PyObject_Call(datetime_class, args, kwargs);
+}
+
+PyObject* py_create_datetime(const ignite::ignite_timestamp &value) {
+    // TODO: Cache classes and functions for re-use
```

Comment:
Ticket for TODO?

Reply:
Good catch. Fixed it already, just forgot to remove TODO.